pkg/kademlia/dialer_test.go

```diff
@@ -152,6 +158,131 @@ func TestDialer(t *testing.T) {
 	})
 }
 
+func TestSlowDialerHasTimeout(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 3,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		// TODO: also use satellites
+		peers := planet.StorageNodes
+
+		{ // PingNode
+			self := planet.StorageNodes[0]
+
+			tlsOpts, err := tlsopts.NewOptions(self.Identity, tlsopts.Config{})
+			require.NoError(t, err)
+
+			self.Transport = transport.NewClient(tlsOpts, 20*time.Millisecond)
+
+			network := &transport.SimulatedNetwork{
+				DialLatency:    200 * time.Second,
+				BytesPerSecond: 1 * memory.KB,
+			}
+
+			slowClient := network.NewClient(self.Transport)
+			require.NotNil(t, slowClient)
+
+			dialer := kademlia.NewDialer(zaptest.NewLogger(t), slowClient)
+			defer ctx.Check(dialer.Close)
+
+			var group errgroup.Group
+			defer ctx.Check(group.Wait)
+
+			for _, peer := range peers {
+				peer := peer
+				group.Go(func() error {
+					_, err := dialer.PingNode(ctx, peer.Local())
+					require.NotNil(t, err)
+					require.Error(t, err, context.DeadlineExceeded)
+					require.True(t, transport.Error.Has(err))
+
+					return nil
+				})
+			}
+		}
+
+		{ // FetchPeerIdentity
+			self := planet.StorageNodes[1]
+
+			tlsOpts, err := tlsopts.NewOptions(self.Identity, tlsopts.Config{})
+			require.NoError(t, err)
+
+			self.Transport = transport.NewClient(tlsOpts, 20*time.Millisecond)
+
+			network := &transport.SimulatedNetwork{
+				DialLatency:    200 * time.Second,
+				BytesPerSecond: 1 * memory.KB,
+			}
+
+			slowClient := network.NewClient(self.Transport)
+			require.NotNil(t, slowClient)
+
+			dialer := kademlia.NewDialer(zaptest.NewLogger(t), slowClient)
+			defer ctx.Check(dialer.Close)
+
+			var group errgroup.Group
+			defer ctx.Check(group.Wait)
+
+			group.Go(func() error {
+				_, err := dialer.FetchPeerIdentity(ctx, planet.Satellites[0].Local())
+				require.NotNil(t, err)
```

Comment:
I believe that you don't have to `require.NotNil` if next you are using `require.Error`

Reply:
oops, that's fair